core/runtime/src/test/java/io/quarkus/runtime/configuration/DotEnvTestCase.java

```diff
@@ -83,7 +84,7 @@ public void testProperties() throws IOException {
                 singletonMap("FOO_BAR", "foo.bar"),
                 singletonMap("foo.baz", "nothing")));
         assertEquals("foo.bar", config.getValue("foo.bar", String.class));
-        assertFalse(config.getOptionalValue("foo.baz", String.class).isPresent());
+        assertTrue(config.getOptionalValue("foo.baz", String.class).isPresent());
```

Comment:
I'm a bit puzzled by this change?

Reply:
Related with https://github.com/quarkusio/quarkus/issues/11782. If someone sets the config key with dot separators in the DotEnv source, we were only looking for the equivalent normalised to upper case and underscore separated. The SR Config DotEnv and Env source also search for an exact match.